File: largestprimefactor.cpp
#include <iostream>
#include <math.h>
using namespace std;

long maxPrime(long n){
    long maxprime=0;
    while(n%2==0){
        maxprime=2;
        n=n/2;
    }
    for(int i=3;i<=sqrt(n);i+=2){
        while(n%i==0){
            maxprime=i;
            n=n/i;
        }
    }
    if(n>2){
        maxprime=n;
    }
    return maxprime;
}

int main() {
	//code
	int t;
	long n;
	cin>>t;
	while(t--){
	    cin>>n;
        cout<<maxPrime(n)<<"\n";
	}
	return 0;
}

File: largestprimefactor.py
#....factors of number n
def getFactors(n):
    
    factors=[]
    for i in range(1, int(math.sqrt(n)+1)):
        if n%i ==0:
            factors.append(i)
            factors.append(n//i)
    # print(factors)
    return factors

#....check if the number are prime
def isPrime(n):
    return len(getFactors(n))==2
    
#main function...to print the largest prime number    
t=int(input())
for _ in range(t):
    n=int(input())
    factors=getFactors(n)
    largestprimefactor=0
    for eachfactor in factors:
        if(isPrime(eachfactor) and eachfactor>largestprimefactor):
            largestprimefactor=eachfactor
    print(largestprimefactor)
